pkg/collection/stack.go2

```diff
@@ -0,0 +1,54 @@
+package collection
+
+
+type Stack[E any] struct {
+	l *LinkedList[E]
+}
+
+func InitStack[E any]() *Stack[E] {
+	return &Stack[E]{
+		l: InitLinkedList[E](),
+	}
+}
+
+func (s *Stack[E]) Len() int {
+	return s.l.len
+}
+
+func (s *Stack[E]) Empty() bool {
+	return s.l.len == 0
+}
+
+//remove all elements of stack
+func (s *Stack[E]) Clear() {
+	s.l.Clear()
+}
+
+func (s *Stack[E]) Traverse() []E {
+	ret := []E{}
+	cur := s.l.head.Next
+	for i := 0; i < s.l.Len(); i++ {
+		ret = append(ret, cur.Val)
+		cur = cur.Next
+	}
+	return ret
+}
```

Comment:
I think `Traverse` should return the slice that it's elements was ordered by `Pop` method.
For example, I run `Push(1)` and `Push(2)` , and then I invoke `Traverse`, I expect the result is `2, 1`.